File: app/src/main/java/se/gustavkarlsson/skylight/android/dagger/modules/definitive/GeomagLocationModule.java
package se.gustavkarlsson.skylight.android.dagger.modules.definitive;

import dagger.Binds;
import dagger.Module;
import dagger.Reusable;
import se.gustavkarlsson.skylight.android.background.providers.GeomagLocationProvider;
import se.gustavkarlsson.skylight.android.background.providers.impl.GeomagLocationProviderImpl;

@Module
public abstract class GeomagLocationModule {

	@Binds
	@Reusable
	abstract GeomagLocationProvider bindGeomagLocationProvider(GeomagLocationProviderImpl impl);

}

File: app/src/main/kotlin/se/gustavkarlsson/skylight/android/dagger/modules/definitive/GeomagLocationModule.kt
package se.gustavkarlsson.skylight.android.dagger.modules.definitive

import dagger.Binds
import dagger.Module
import dagger.Reusable
import se.gustavkarlsson.skylight.android.services.providers.GeomagLocationProvider
import se.gustavkarlsson.skylight.android.services_impl.providers.GeomagLocationProviderImpl

@Module
abstract class GeomagLocationModule {

    @Binds
    @Reusable
    abstract fun bindGeomagLocationProvider(impl: GeomagLocationProviderImpl): GeomagLocationProvider

}
